http/codegen/openapi/v3/builder.go

```diff
@@ -210,21 +210,37 @@ func buildOperation(key string, r *expr.RouteExpr, bodies *EndpointBodies, rand
 
 	// request body
 	var requestBody *RequestBodyRef
-	if e.Body.Type != expr.Empty {
+	{
 		ct := "application/json" // TBD: need a way to specify method media type in design...
-		if e.MultipartRequest {
-			ct = "multipart/form-data"
-		}
-		mt := &MediaType{
-			Schema:  bodies.RequestBody,
-			Example: e.Body.Example(rand),
-		}
-		requestBody = &RequestBodyRef{Value: &RequestBody{
-			Description: e.Body.Description,
-			Required:    e.Body.Type != expr.Empty,
-			Content:     map[string]*MediaType{ct: mt},
-			Extensions:  openapi.ExtensionsFromExpr(e.Body.Meta),
-		}}
+		if e.Body.Type != expr.Empty {
+			if e.MultipartRequest {
+				ct = "multipart/form-data"
+			}
+			mt := &MediaType{
+				Schema:  bodies.RequestBody,
+				Example: e.Body.Example(rand),
+			}
+			requestBody = &RequestBodyRef{Value: &RequestBody{
+				Description: e.Body.Description,
+				Required:    e.Body.Type != expr.Empty,
+				Content:     map[string]*MediaType{ct: mt},
+				Extensions:  openapi.ExtensionsFromExpr(e.Body.Meta),
+			}}
+		} else if e.SkipRequestBodyEncodeDecode {
+			// it is possible for design to have request Body as Empty and set
```

Comment:
I don't understand this case. If SkipXXX is used then it seems that the OpenAPI request body schema should be computed by looking up all the attributes of the payload that are not used in headers, cookies and params? If the body type is set to empty then maybe we should honor that in the OpenAPI schema?